src/battlearena/BattleArena.js

```diff
@@ -4,36 +4,45 @@ import * as React from 'react';
 import { Link } from 'react-router-dom';
 import Navbar from '../components/Navbar.js';
 import Leaderboard from '../components/Leaderboard.js';
+import RenderTank from '../components/RenderTank.js';
+import SearchPlayers from '../components/SearchPlayers.js';
+
+// searchName tracks the text the user enters into the searchbar to find players to challenge.
+type State = {|
+	searchName: string
+|};
+
+class BattleArena extends React.Component<State> {
```

Comment:
Unless something very confusing is going on here that I'm not aware of, you are actually specifying the type of your Props as the type called 'State'. You need to define the type of your props before the state. See this for docs: https://flow.org/en/docs/react/components/